pkg/k8s/volumes/crud.go

```diff
@@ -71,12 +71,22 @@ func CreateForDev(ctx context.Context, dev *model.Dev, c *kubernetes.Clientset,
 		pvc.Spec.VolumeName = k8Volume.Spec.VolumeName
 		_, err = vClient.Update(ctx, pvc, metav1.UpdateOptions{})
 		if err != nil {
-			return fmt.Errorf("error updating kubernetes volume claim: %s", err)
+			if !isDynamicallyProvisionedPVCError(err, pvc.Name) {
+				return fmt.Errorf("error updating kubernetes volume claim: %s", err)
+			}
+			oktetoLog.Debug("could not update pvc in namespace: %w", dev.Namespace, err)
+			oktetoLog.Warning("Error while updating the dev volume: try running 'okteto down -v' and 'okteto up', or talk to your administrator (the PVC's storage class must support 'allowVolumeExpansion' to be able to upscale dev volumes)")
```

Comment:
If I understood correctly the ticket, it says that the volume has to be recreated for deployments when the update fails. Where are we recreating the volume? I don't see it

Reply:
We had a conversation with @pchico83 about the UX for the migration from 2Gi to 5Gi and we settled for only showing a warning if the update failed due to the upscaling (I should've written it down, sorry). 

Reading this again, I think a better UX would be to always handle the error and recreate the volume (as the [comment in the issue](https://github.com/okteto/okteto/issues/3045#issuecomment-1245857059) states). @pchico83 why did we change to showing only a warning when we couldn't upscale?

Reply:
Ok, then, I think the message can be confusing for the user. We are printing a Warning message that start with `Error` and we are not breaking the execution returning an error, so I guess the process just keeps working but the volume size is the old one.

If that is the case, I think that the message should say something like `Could not increase the size of the dev volume: try running 'okteto down -v' and 'okteto up', or talk to your administrator (the PVC's storage class must support 'allowVolumeExpansion' to be able to upscale dev volumes). Previous size %d is being used` (or something similar). Maybe @RinkiyaKeDad can help with the text.

One concern about this is that we could start prompting this message automatically to some users that are working fine with the CLI after a CLI update. @pchico83 Is it a common case that enterprise users use default values? or this change is only to avoid friction to people starting to use Okteto?

Reply:
:+1: It makes sense to not say `Error` when we are throwing a warning